hudi-common/src/main/java/org/apache/hudi/avro/HoodieAvroUtils.java

```diff
@@ -427,19 +429,138 @@ public static Schema getRecordKeyPartitionPathSchema() {
   }
 
   /**
-   * Fetch schema for record key and partition path.
+   * Fetches projected schema given list of fields to project. The field can be nested in format `a.b.c` where a is
+   * the top level field, b is at second level and so on.
    */
   public static Schema getSchemaForFields(Schema fileSchema, List<String> fields) {
-    List<Schema.Field> toBeAddedFields = new ArrayList<>();
-    Schema recordSchema = Schema.createRecord("HoodieRecordKey", "", "", false);
+    List<LinkedList<String>> fieldPathLists = new ArrayList<>();
+    for (String path : fields) {
+      fieldPathLists.add(new LinkedList<>(Arrays.asList(path.split("\\."))));
+    }
 
-    for (Schema.Field schemaField : fileSchema.getFields()) {
-      if (fields.contains(schemaField.name())) {
-        toBeAddedFields.add(new Schema.Field(schemaField.name(), schemaField.schema(), schemaField.doc(), schemaField.defaultVal()));
+    Schema result = Schema.createRecord("HoodieRecordKey", "", "", false);
```

Comment:
we should fix the record name, doc and namespace to be same with the `fileSchema` ?

Reply:
Addressed